### DISCUSSION TITLE
Authelia server does not go back to Client callback url which is mentioned in redirect_uris

### DISCUSSION THREAD

**Discussion by shailendra4dev** [authelia/authelia#5130] [Q&A]
Dear Team,

I am implementing Authelia as an OIDC provider for my node js client application. 

It does not go back to the client callback url (redirect_uris) instead it is redirecting to default_redirection_url, here is my Authelia configuration code. Any help would be highly appreciated. Thanks.

jwt_secret: RcmxfUOBEPme0bkAkRtyrLo1szLM8zaQz8KkqvP7UmUg4rlAOzf4AKZ7xePeCART
      server:
        host: 0.0.0.0
        port: ${bind=authelia:port[service]}
      log:
        level: debug
      default_redirection_url: https://client-app.test.com
      authentication_backend:
        file:
          path: /config/users_database.yml
      access_control:
        default_policy: deny
        rules:
          - domain:
              - client-app.test.com
            policy: one_factor
      session:
        name: authelia_session
        secret: RcmxfUOBEPme0bkAkRtyrLo1szLM8zaQz8KkqvP7UmUg4rlAOzf4AKZ7xePeZXCV
        expiration: 12h
        inactivity: 5m
        remember_me_duration: 1M
        domain: client-app.test.com
        redis:
          host: ${bind=authelia-redis}
          port: ${bind=authelia-redis:port[service]}
      regulation:
        max_retries: 3
        find_time: 5m
        ban_time: 15m
      storage:
        encryption_key: RcmxfUOBEPme0bkAkRtyrLo1szLM8zaQz8KkqvP7UmUg4rlAOzf4AKZ7xePeLKJH
        postgres:
          host: ${bind=authelia-psql}
          port: ${bind=authelia-psql:port[db]}
          database: ${bind=authelia-psql:var:POSTGRES_DB}
          schema: public
          username: ${bind=authelia-psql:var:POSTGRES_USER}
          password: ${bind=authelia-psql:var:POSTGRES_PASSWORD}
          timeout: 5s
      notifier:
        disable_startup_check: true
        smtp:
          username: test@gmail.com
          password: <gmail_password>
          host: smtp.gmail.com
          port: 587
          sender: test@gmail.com
      identity_providers:
        oidc:
          hmac_secret: RcmxfUOBEPme0bkAkRtyrLo1szLM8zaQz8KkqvP7UmUg4rlAOzf4AKZ7xePeQWER
          issuer_private_key: |
            -----BEGIN RSA PRIVATE KEY-----
            PRIVATE KEY
            -----END RSA PRIVATE KEY-----
          access_token_lifespan: 1h
          authorize_code_lifespan: 1m
          id_token_lifespan: 1h
          refresh_token_lifespan: 90m
          enable_client_debug_messages: false
          cors:
            endpoints:
              - authorization
              - token
              - revocation
              - introspection
              - userinfo
            allowed_origins:
              - '*'
            allowed_origins_from_client_redirect_uris: false
          clients:
            - id: nodeapp
              description: Node Application
              secret: mysecret
              sector_identifier: ''
              public: false
              authorization_policy: one_factor
              consent_mode: explicit
              pre_configured_consent_duration: 1w
              audience: []
              scopes:
                - openid
                - groups
                - email
                - profile
              redirect_uris:
                - https://client-app.test.com/login/callback
              grant_types:
                - refresh_token
                - authorization_code
              response_types:
                - code
              response_modes:
                - form_post
                - query
                - fragment
              userinfo_signing_algorithm: none

**Comment by james-d-elliott** [comment]
Please show how you're invoking the authorization request

**Reply by shailendra4dev** [reply]
It is the Client App code.

`process.env["NODE_TLS_REJECT_UNAUTHORIZED"] = 0;

const express = require('express');
const session = require('express-session');
const cookieParser = require('cookie-parser');
const helmet = require('helmet');
const passport = require('passport');

const http    = require("http");

const { Issuer,Strategy } = require('openid-client');

const port = process.env.CLIENT_APP_PORT;

const app = express();

app.use(cookieParser());
app.use(express.urlencoded({
  extended: true,
}));


app.use(express.json({ limit: '15mb' }));
app.use(session({secret: 'secret', 
                 resave: false, 
                 saveUninitialized: true,}));
app.use(helmet());
app.use(passport.initialize());
app.use(passport.session());

passport.serializeUser(function(user, done) {
    console.log('-----------------------------');
    console.log('serialize user');
    console.log(user);
    console.log('-----------------------------');
    done(null, user);
});
passport.deserializeUser(function(user, done) {
    console.log('-----------------------------');
    console.log('deserialize user');
    console.log(user);
    console.log('-----------------------------');
    done(null, user);
});

Issuer.discover(process.env.OAUTH_URI)
  .then(function (oidcIssuer) {
    var client = new oidcIssuer.Client({
      client_id: process.env.CLIENT_ID,
      client_secret: process.env.CLIENT_SECRET,
      redirect_uris: [process.env.CLIENT_REDIRECT_URIS],
      scope: process.env.CLIENT_SCOPE,
      response_types: [process.env.CLIENT_RESPONSE_TYPES],
      response_mode: process.env.CLIENT_RESPONSE_MODE
    });

    passport.use(
      'oidc',
      new Strategy({ client,passReqToCallback: true}, (req, tokenSet, userinfo, done) => {
        console.log("tokenSet",tokenSet);
        console.log("userinfo",userinfo);
        req.session.tokenSet = tokenSet;
        req.session.userinfo = userinfo;
        return done(null, tokenSet.claims());
      }
      
      )
    );
  });

app.get('/login',
function (req, res, next) {
    console.log('-----------------------------');
    console.log('/Start login handler');
    next();
},
passport.authenticate('oidc',{scope:"openid"}));

app.get('/login/callback',(req,res,next) =>{
  console.log('-----------------------------');
  console.log('/login/callback');
  passport.authenticate('oidc',{ successRedirect: '/user',
  failureRedirect: '/' })(req, res, next)
}
  
)

app.get("/",(req,res) =>{
   let url = process.env.OAUTH_URI;
   res.send(" <a href=" + url + ">Authelia Server </a>")
})
app.get ("/user",(req,res) =>{
    res.header("Content-Type",'application/json');
    res.end(JSON.stringify({tokenset:req.session.tokenSet,userinfo:req.session.userinfo},null,2));

})

http.createServer(app)
.listen(port, function (req, res) {
  console.log("Server started at port " + port);
});`

**Reply by james-d-elliott** [reply]
Can you show the authelia trace/debug logs when performing these requests and provide the version you're using? Also what does `process.env.OAUTH_URI` contain? You can redact the public portion of the domain.

**Reply by james-d-elliott** [reply]
@shailendra4dev still interested in the debug/trace logs for authelia. It should show what's going on. In addition consider a [HAR](https://www.authelia.com/reference/guides/http-archive-file/)
